src/votypes.jl

```diff
@@ -58,9 +61,10 @@ function jl2votype(T::Type)
 end
 
 function vo2nbytes_fixwidth(colspec)
-    get(colspec, :arraysize, nothing) == "*" && return nothing
-    N = get(colspec, :arraysize, 1)
-    return N * TYPE_VO_TO_NBYTES[colspec[:datatype]]
+    arraysize = get(colspec, :arraysize, "1")
+    arraysize[end] == '*' && return nothing
```

Comment:
No, because variable-length arrays can be multidimensional.  As per VOTable 1.4 sec 2.2, `arraysize` values like "`3x4x*`" or "`100x20*`" are permitted.  Any array with a variable length (which can only be in the final dimension) will (a) end in a `*` character and (b) have no fixed length, hence will need an element count in the BINARY/BINARY2 stream.